daffodil-test/src/test/scala/org/apache/daffodil/layers/TestBad.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.daffodil.layers
+
+import org.junit.AfterClass
+import org.junit.Assert._
+import org.junit.Test
+import org.apache.daffodil.tdml.{Runner, TDMLException}
+
+object TestBad {
+
+  lazy val runner = Runner("/org/apache/daffodil/layers", "TestBad.tdml")
+
+  @AfterClass def shutDown(): Unit = {
+    runner.reset
+  }
+}
+
+class TestBad {
+
+  import TestBad._
+
+  @Test def test_bad(): Unit = {
+    try {
+      runner.runOneTest("test_bad")
+      assert(false, "expected exception but none thrown")
+    } catch {
+      case te: TDMLException =>
+        assertNotNull(te.getCause)
+        assertEquals("org.apache.daffodil.layers.LayerExecutionException", te.getCause.getClass.getName)
+      case e: Exception => {
+        assert(false, s"expected TDMLException but got ${e.getClass.getName}")
+      }
```

Comment:
Instead of this match/case, you can use `intercept` to expect a certain type of exception. If it doesn't match or an exception isn't throw then an error is thrown, so you don't have to worry about the non-TDMLException case or do any try/catch stuff.

Reply:
This test was removed in favor of the test moved over to daffodil-cli.